docs/command-completion.md

```diff
@@ -0,0 +1,30 @@
+Command Completion
+==================
+
+Shell command completion is provided by the script at 
+`/misc/completion/ipfs-completion.bash`.
+
+
+Installation
+------------
+The simplest way to see it working is to run 
+`source misc/completion/ipfs-completion.bash` straight from your shell. This
+is only temporary and to fully enable it, you'll have to follow one of the steps
+below.
+
+### Bash on Linux
+For bash, completion can be enabled in a couple of ways. One is to add
+```bash
+if [ -n "$GOPATH" ]; then
+  source $GOPATH/src/github.com/ipfs/go-ipfs/misc/completion/ipfs-completion.bash
+fi
+```
```

Comment:
should tell the user what to do if they _dont_ have go. it's possible that we could make a simple `ipfs-install-bash-completion` tool. i want to start making tools that we can deploy through ipfs itself, whose hashes are hardcoded in the binary.